tensorflow/core/grappler/optimizers/generic_layout_optimizer.cc

```diff
@@ -109,10 +109,17 @@ inline std::pair<string, string> GetSrcAndDstDataFormats(
     const TransposeContext& context, int num_gpus, int num_voltas) {
   string src_format = kNHWC;
   string dst_format = kNCHW;
-  if (((static_cast<float>(num_voltas) / static_cast<float>(num_gpus)) >=
-       kVoltaGPURatioThreshold) &&
-      NumConvOnDeviceWithDataTypeOverThreshold(context, kGPU, DT_HALF)) {
-    std::swap(src_format, dst_format);
+  if (!context.enforced_layout.empty()) {
+    // Otherwise, enforced_layout is NCHW enforced by previous check.
+    if (context.enforced_layout == "NHWC") {
+      std::swap(src_format, dst_format);
+    }
+  } else {
+    if (((static_cast<float>(num_voltas) / static_cast<float>(num_gpus)) >=
+        kVoltaGPURatioThreshold) &&
+        NumConvOnDeviceWithDataTypeOverThreshold(context, kGPU, DT_HALF)) {
+      std::swap(src_format, dst_format);
+    }
   }
```

Comment:
We can combine these two branches:
// We enforce enforced_layout is with NCHW or NHWC.
if ( (!context.enforced_layout.empty() && context.enforced_layout == "NHWC") || ...) {
   std::swap(...)
}

Reply:
I wrote it as this this as I thought it be more readable and it would be optimized out
But sure, I can rewrite as you suggest!